### PR TITLE
Added UI test for deleting a puppet repo associated with a content view

### PR DESCRIPTION
**Pull request by sthirugn**
BZ #1271000

### CODE REVIEW

File: tests/foreman/ui/test_repository.py [hunk 1]
```diff
@@ -334,6 +334,29 @@ def test_positive_delete(self):
...
+        2. Create a new puppet repo, no sync source
+        3. Upload a puppet module (say ntpd) to repo
+        4. Create a CV, go to add puppet modules page
+        5. Add latest version of the puppet module from Step 3
+        6. View puppet repo details, it should show "Latest (Currently X.Y.Z)"
+        7. Go back to product, drill down into repo and delete the puppet
+        module from Step 3
+        8. Go back to same CV puppet module details page
+
+        @Assert: Proper error message saying that the puppet module version is
```
Comment (oshtaier) [inline tests/foreman/ui/test_repository.py]: nitpick: 'the puppet module version is not found' is not so obvious. 1271000 BZ has a little bit better message idea in expected results.

**Comment by oshtaier** [comment]
ACK pending non-blocking comment

File: tests/foreman/ui/test_repository.py (hunk 1 above)
Comment (sthirugn) [inline tests/foreman/ui/test_repository.py]: I am not sure I understand your question.  But did you see https://bugzilla.redhat.com/show_bug.cgi?id=1271000#c11? The bug says that after Step 8, they see just a spinner and the page never loads.   Now an error message is getting implemented which would avoid the spinner and the page will load with error message.

Comment (oshtaier) [inline tests/foreman/ui/test_repository.py]: Nope. I didn't read all the comments. Sorry

File: tests/foreman/ui/test_repository.py [hunk 2]
```diff
@@ -334,6 +334,29 @@ def test_positive_delete(self):
...
     @run_only_on('sat')
+    @stubbed
+    @tier2
+    def test_negative_delete_puppet_repo_associated_with_cv(self):
+        """Delete a puppet repo associated with a content view - BZ#1271000
+
+        @Feature: Content Product - Delete repo
+
+        @Steps:
+        1. Create a new product
```
Comment (elyezer) [inline tests/foreman/ui/test_repository.py]: Please add a blank line above to ensure better presentation for the generated HTML on the docs.

File: tests/foreman/ui/test_repository.py [hunk 3]
```diff
@@ -334,6 +334,29 @@ def test_positive_delete(self):
...
+        4. Create a CV, go to add puppet modules page
+        5. Add latest version of the puppet module from Step 3
+        6. View puppet repo details, it should show "Latest (Currently X.Y.Z)"
+        7. Go back to product, drill down into repo and delete the puppet
+        module from Step 3
+        8. Go back to same CV puppet module details page
+
+        @Assert: Proper error message saying that the puppet module version is
+        not found
+        """
```
Comment (elyezer) [inline tests/foreman/ui/test_repository.py]: Missing manual status on the docstring since it is a stub.

**Comment by elyezer** [comment]
ACK, pending comments.

File: tests/foreman/ui/test_repository.py (hunk 3 above)
Comment (sthirugn) [inline tests/foreman/ui/test_repository.py]: :+1:

Comment (sthirugn) [inline tests/foreman/ui/test_repository.py]: I can't believe I missed these :) Thank you for pointing out.

**Comment by sthirugn** [comment]
@elyezer comments addressed and rebased.

**Comment by elyezer** [comment]
APT

**Comment by oshtaier** [comment]
APT

**Pull request merged**